ExtraDry/ExtraDry.Core/Server/Internal/LinqBuilder.cs

```diff
@@ -50,16 +50,16 @@ public static IOrderedQueryable<T> ThenByDescending<T>(this IOrderedQueryable<T>
         /// <remarks>see https://stackoverflow.com/questions/41244/dynamic-linq-orderby-on-ienumerablet-iqueryablet</remarks>
         private static IOrderedQueryable<T> ApplyOrder<T>(IQueryable<T> source, string property, OrderType methodType)
         {
+            var modelDescription = new ModelDescription(typeof(T));
             string[] props = property.Split('.');
             var type = typeof(T);
             var arg = Expression.Parameter(type, "x");
             Expression expr = arg;
             foreach(string prop in props) {
-                // use reflection (not ComponentModel) to mirror LINQ
-                var pi = type.GetProperty(prop, BindingFlags.Public | BindingFlags.IgnoreCase | BindingFlags.Instance);
-                if(pi == null) {
-                    throw new DryException($"Could not find sort property `{prop}`", "Could not apply requested sort");
-                }
+                var pi = modelDescription.SortProperties.FirstOrDefault(sortProp => sortProp.ExternalName.ToLower() == prop.ToLower())?.Property
+                    ?? (modelDescription.StabilizerProperty.ExternalName.ToLower() == prop.ToLower() 
```

Comment:
I don't do this consistently enough but this comparison requires two memory allocations.   Preferred alternative is String.Compare(a, b, StringComparisonOptions.IgnoreCase)

Reply:
Updated to use `string.Compare`.